### PR TITLE
User guide update

### LINKED ISSUE #279: Update match command for UG

### PR DESCRIPTION
**Pull request by keanecjy**
PR Overview:

- Add find command format for application
- Add match command images and explanation

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -532,15 +537,40 @@ Format: `list me`
...
-- Internships with job titles matching at least one `SKILL` will be returned.
-- Only full words will be matched. e.g. `React` will not match `React Native`
+- Only the `Requirement` field in the internships will be searched for.
+- Only profile items with `Skill` category will be used for matching.
+- An Internship with at least one `Requirement` matching any one `Skill` in the profile list will be
+considered as a successful match. e.g. Say you have a profile item which is of `Skill` category and titled `Python`, any
+internship that has a `Requirement` of `Python` will be successfully matched.
+- Matched internships can have `Requirements` that do not match with the profile `Skills`, since a match is found when
+an internship has **at least one and not all** `Requirements` that matches with the profile `Skills`.
+- Matching done is case-insensitive. e.g. `Python` will match `python`.
```
Comment (ZoroarkDarkrai) [inline docs/UserGuide.md:546]: Matching done is a bit weird(?)
Matching is case-insensitive?
Matching is done case insensitively?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -532,15 +537,40 @@ Format: `list me`
...
+- Only the full sentence will be matched. e.g. `Machine` will not match `Machine Learning`.
```
Comment (ZoroarkDarkrai) [inline docs/UserGuide.md]: phrase rather than sentence?

File: docs/UserGuide.md [hunk 3]
```diff
@@ -398,12 +398,17 @@ Executing `view app 2` will update the right panel to display the second applica
 Finds all applications in your list of applications whose job titles contain any of the given keywords.
 
 Format: `find app KEYWORD [ANOTHER_KEYWORD]...`
-- Find command matching is the same as that used in [`find com` command](#finding-companies-find-com), except 
-job titles are used in placed of company names.
+- Only the application name will be searched for.
```
Comment (ZoroarkDarkrai) [inline docs/UserGuide.md]: application job title?

File: docs/UserGuide.md [hunk 4]
```diff
@@ -398,12 +398,17 @@ Executing `view app 2` will update the right panel to display the second applica
...
+- The search is case-insensitive. e.g. `engineer` will match `Engineer`.
+- Companies with names matching at least one keyword will be returned. e.g. `Software Engineer` will return applications
```
Comment (ZoroarkDarkrai) [inline docs/UserGuide.md]: Applications?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -521,13 +521,13 @@ This is how the `MatchCommand#execute()` method works upon execution:
 `MatchCommand`'s own `getSkillList(...)` method.
 3. Next, the list of all internships from the list of companies is obtained via its own
 `getInternshipList(...)` method.
-4. Then, the `MatchCommand`'s own `getMatchingInternships(...)` is invoked to obtain
+4. Then, `MatchCommand`'s own `getMatchingInternships(...)` is invoked to obtain
```
Comment (ZoroarkDarkrai) [inline docs/DeveloperGuide.md]: why not `MatchCommand#getMatchingInternships(...)`?

**Review by ZoroarkDarkrai — COMMENTED** [review]
LGTM, just a few nitpicks

File: docs/UserGuide.md (hunk 1 above)
Comment (keanecjy) [reply docs/UserGuide.md:546]: Okay, will update

File: docs/UserGuide.md (hunk 2 above)
Comment (keanecjy) [reply docs/UserGuide.md]: Thanks, been finding a way to make this clearer

File: docs/UserGuide.md (hunk 3 above)
Comment (keanecjy) [reply docs/UserGuide.md]: Okay will update

**Review by orzymandias — APPROVED** [review]

**Pull request merged**